### ISSUE TITLE
Clean up VirtualEnv API

### ISSUE THREAD

**Issue by nickjalbert** [agentos-project/agentos#297]
I started some clean up work on the VirtualEnv API in #296, but we can probably make it better.  In general, I imagine the primary public interface could look like the following:

```python

class VirtualEnv:
    def activate(self):
         pass

    def deactivate(self):
         pass

    def install_requirements_file(self, file_path):
         pass

    # To support setup.py ala #275 
    def install_requirement(self, requirement_name):
         pass
```

And then some bonus class methods and helper methods.  I don't think we're too far from this after #296 is merged, but it will probably be good to do another pass before tackling #275.

**Comment by andyk** [comment]
@nickjalbert is this done now as of #306 being merged which addressed #275? I think that at least `activate()` and `deactivate()` now are there, not sure if you have other changes you want to make to the way we are handling requirements files?

**Comment by nickjalbert** [comment]
Yeah, let's close this for now.  I still think there's some work to do on `virtual_env.py`, but nothing super low-hanging.
